src/shared/components/oncoprint/controls/OncoprintControls.tsx

```diff
@@ -208,6 +211,18 @@ export default class OncoprintControls extends React.Component<IOncoprintControl
         this.onHorzZoomSliderSet = this.onHorzZoomSliderSet.bind(this);
         this.onSetHorzZoomTextInput = this.onSetHorzZoomTextInput.bind(this);
 
+        // add pre-selected treatments (passed in via URL in ResultsViewOncoprint) to selected treatments
+        this.initSelectedTreatments = autorun(() => {
+            const selectionMap = this.treatmentOptionsByValueMap;
+            if (_.keys(selectionMap).length > 0) {
+                const initTreatmentOpts = this.props.state.selectedTreatmentsInit ?
+                    this.props.state.selectedTreatmentsInit.map((treatmentId) => selectionMap[treatmentId])
```

Comment:
minor but you might want to reference the props which passed to contructor here instead of this.props, which i think wouldn't exist were this callback to fire immediately (syncronously).   props is available here via closure

Reply:
Updated